hera_cal/nucal.py

```diff
@@ -0,0 +1,473 @@
+    radially redundant groups based on a number of factors, can get specific polarizations, and
+    radially redundant and spatially redundant groups by baseline key.
+    """
+
+    def __init__(
+        self, antpos, reds=None, blvec_error_tol=1e-9, pols=["nn"], bl_error_tol=1.0
+    ):
+        """
+        Parameters:
+        ----------
+        antpos : dict
+            Antenna positions in the form {ant_index: np.array([x,y,z])}.
+        reds : list of list
+            List of lists of baseline keys. Can be determined using redcal.get_reds
+        pols : list of strs
+            List of polarization strings to be used in the frequency redundant group
+        """
+        self.antpos = antpos
+
+        if reds is None:
+            reds = redcal.get_reds(antpos, pols=pols, bl_error_tol=bl_error_tol)
+
+        self._mapped_reds = {red[0]: red for red in reds}
+
+        self._radial_groups = get_unique_orientations(
+            antpos, reds=reds, pols=pols, blvec_error_tol=blvec_error_tol
+        )
+
+    def get_radial_group(self, key):
+        """
+        Get baselines with the same radal heading as a given baseline
+
+        Parameters:
+        ----------
+        key : tuple
+            Baseline key of type (ant1, ant2, pol)
+
+        Returns:
+        -------
+        group : list of tuples
+            List of baseline tuples that have the same radial headings
+
+        """
+        # Identify headings
+        for group_key in self._mapped_reds:
+            if key == group_key or key in self._mapped_reds[group_key]:
+                key = group_key
+                break
+
+        for group in self._radial_groups:
+            if key in group:
+                return group
+
+    def get_redundant_group(self, key):
+        """
+        Get a list of baseline that are spatially redundant with the input baseline
+
+        Parameters:
+        ----------
+        key: tuple
+            Baseline key with of type (ant1, ant2, pol)
+
+        Returns:
+        -------
+        group: list of tuples
+            Return baseline tuples that are spatially redundant
+        """
+        for group_key in self._mapped_reds:
+            if key == group_key or key in self._mapped_reds[group_key]:
+                key = group_key
+                break
+
+        if utils.reverse_bl(key) in self._mapped_reds:
+            return [
+                utils.reverse_bl(bls)
+                for bls in self._mapped_reds.get(utils.reverse_bl(key))
+            ]
+        elif key in self._mapped_reds:
+            return self._mapped_reds.get(key)
+        else:
+            raise KeyError(
+                f"Baseline {key} is not in the group of spatial redundancies"
+            )
+
+    def get_pol(self, pol):
+        """Get all radially redundant groups with a given polarization"""
+        for group in self:
+            if group.pol == pol:
+                yield group
```

Comment:
Is there a reason this is an iterator? It's useful when you're doing loops, but it's a bit weird for a funciton called `get_pol` instead of `iterate_over_pols` or similar.

Reply:
Changed this so it just returns the list of baseline groups that have a give polarization